content/5_Plat/RURQ.mdx

```diff
@@ -148,139 +148,138 @@ situations might take place after any update:
   $\texttt{lzSet}$ value since a set update will override all previous add
   updates.
 
-Given the mechanics behind the `pushdown` function, all you need is a regular
+Given the mechanics behind the `push_down` function, all you need is a regular
 range-sum segment tree to solve the question.
 
 ```cpp
-
-#include <iostream>
-using ll = long long;
+#include <bits/stdc++.h>
 using namespace std;
+using ll = long long;
 
-const int maxN = 2e5 + 5;
-
-int N, Q;
-int a[maxN];
+/**
+ * Represents the type of lazy update being done.
+ * NONE = if there's no lazy update to be performed.
+ */
+enum QueryType { ADD, SET, NONE };
 
-struct node {
-	ll val;
-	ll lzAdd;
-	ll lzSet;
-	node(){};
-} tree[maxN << 2];
+struct Query {
+	QueryType type = NONE;
+	ll val = 0;
+};
 
-#define lc p << 1
-#define rc (p << 1) + 1
+/**
+ * T = tree node type, which will be long long
+ * U = update type, which will be the Query struct
+ */
+template <class T, class U> class LazySegtree {
```

Comment:
is `U` still necessary?

Reply:
i could get rid of the template stuff if u want

Reply:
uh just `U`, you can keep `T`

Reply:
kk gimme a sec